### ISSUE TITLE
Error when deleting (Error: Namespace Resolver for 'Object' not found! This is probably a bug in '@amplify-js/datastore'.)

### ISSUE THREAD

**Issue by madmaxaus** [aws-amplify/amplify-js#10866]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React, Next.js

### Amplify APIs

DataStore

### Amplify Categories

Not applicable

### Environment information

<details>

```
# Put output below this line

  System:
    OS: Windows 10 10.0.22000
    CPU: (16) x64 AMD Ryzen 7 Microsoft Surface (R) Edition
    Memory: 3.46 GB / 15.45 GB
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (108.0.1462.76)
    Internet Explorer: 11.0.22000.120
  npmGlobalPackages:
    @aws-amplify/cli: 10.0.0
    amplify: 0.0.11
    azure-functions-core-tools: 4.0.4915
    yarn: 1.22.19
```

</details>

### Describe the bug

When i try to delete a record using datastore i get the error: (other non-delete queries work)

        
       Error: Namespace Resolver for 'Object' not found! This is probably a bug in '@amplify-js/datastore'.
    at IndexedDBAdapter.namespaceResolver (datastore.js?38c2:94:1)
    at IndexedDBAdapter.getIndexKeyValuesFromModel (IndexedDBAdapter.js?e5aa:38:1)
    at IndexedDBAdapter.eval (IndexedDBAdapter.js?e5aa:1031:1)
    at step (tslib.es6.js?6654:100:1)
    at Object.eval [as next] (tslib.es6.js?6654:81:46)
    at fulfilled (tslib.es6.js?6654:71:43)

### Expected behavior

Delete the object like it used to

### Reproduction steps

1.  run dev (yarn dev)
2. visit my page
3. click the delete button (and confirm)
4. doesn't delete throws error

### Code Snippet

```javascript
// Put your code below this line.
            console.log("EDITING TEST RESULT", testResultID);
            const toDelete = await DataStore.query(TestResult, testResultID);
            console.log("TO DELETE", toDelete);
            DataStore.delete(toDelete);
```

### Log output

<details>

```
// Put your logs below this line
       Error: Namespace Resolver for 'Object' not found! This is probably a bug in '@amplify-js/datastore'.
    at IndexedDBAdapter.namespaceResolver (datastore.js?38c2:94:1)
    at IndexedDBAdapter.getIndexKeyValuesFromModel (IndexedDBAdapter.js?e5aa:38:1)
    at IndexedDBAdapter.eval (IndexedDBAdapter.js?e5aa:1031:1)
    at step (tslib.es6.js?6654:100:1)
    at Object.eval [as next] (tslib.es6.js?6654:81:46)
    at fulfilled (tslib.es6.js?6654:71:43)

```

</details>

### aws-exports.js

/* eslint-disable */
// WARNING: DO NOT EDIT. This file is automatically generated by AWS Amplify. It will be overwritten.

const awsmobile = {
    "aws_project_region": "ap-southeast-2",
    "aws_cloud_logic_custom": [
        {
            "name": "AdminQueries",
            "endpoint": "https://fxn1bcqlef.execute-api.ap-southeast-2.amazonaws.com/dev",
            "region": "ap-southeast-2"
        },
        {
            "name": "portalRESTAPI",
            "endpoint": "https://v723682xa1.execute-api.ap-southeast-2.amazonaws.com/dev",
            "region": "ap-southeast-2"
        }
    ],
    "aws_appsync_graphqlEndpoint": "https://ky4kljscojeodctanqtavqce5q.appsync-api.ap-southeast-2.amazonaws.com/graphql",
    "aws_appsync_region": "ap-southeast-2",
    "aws_appsync_authenticationType": "AMAZON_COGNITO_USER_POOLS",
    "aws_appsync_apiKey": "doYouNeedThis?",
    "aws_cognito_identity_pool_id": "ap-southeast-2:a666fa81-0590-46c4-b044-38251abd0876",
    "aws_cognito_region": "ap-southeast-2",
    "aws_user_pools_id": "ap-southeast-2_xvmBUxeBk",
    "aws_user_pools_web_client_id": "2kc8pjrdc6jodj085g4sc5fpvk",
    "oauth": {},
    "aws_cognito_username_attributes": [
        "EMAIL"
    ],
    "aws_cognito_social_providers": [],
    "aws_cognito_signup_attributes": [
        "EMAIL",
        "NAME"
    ],
    "aws_cognito_mfa_configuration": "OPTIONAL",
    "aws_cognito_mfa_types": [
        "TOTP"
    ],
    "aws_cognito_password_protection_settings": {
        "passwordPolicyMinLength": 8,
        "passwordPolicyCharacters": []
    },
    "aws_cognito_verification_mechanisms": [
        "EMAIL"
    ],
    "aws_user_files_s3_bucket": "portal-user-files145131-dev",
    "aws_user_files_s3_bucket_region": "ap-southeast-2"
};

export default awsmobile;

### Additional information and screenshots

```
type TestResults @model @auth(rules: [{allow: public}, {allow: owner, operations: [create, read, update]}, {allow: groups, groups: ["admin"], operations: [read, create, update, delete]}]) {
  id: ID!
  CustomerID: ID! @index(name: "byCustomer")
  readUsers: [String]
  readGroups: [String]
  editUsers: [String]
  editGroups: [String]
  lab: String
  fasting: Boolean
  dateReported: AWSDateTime
  dateCollected: AWSDateTime
  dateTested: AWSDateTime
  type: [TestType]
  note: String
  UploadedFiles: [UploadedFiles] @hasMany(indexName: "byTestResults", fields: ["id"])
  TestResultData: [TestResultData] @hasMany(indexName: "byTestResults", fields: ["id"])
  DNATestResults: [DNATestResult] @hasMany(indexName: "byTestResults", fields: ["id"])
}
```

This only just started happening after i pulled an update aws-amplify module. It used to work perfectly.

**Comment by bobalucard** [comment]
I have just started having this issue after updating @amplify-cli from v8 to v10.6. It is only happening on a specific model and when I try to delete. All other models are fine. I have changed the model schema and repushed using amplify-cli which did not fix the issue. The below is my model:

`type Project @model @auth(rules: [
    { allow: groups, groups: ["admin","facilitator","client"]},
    { allow: owner, ownerField: "staff" }
  ]) {
  id: ID!
  businessID: ID @index(name: "byBusiness")
  owner: String
  staff: [String]
  name: String
  description: String
  methodology: Methodology @hasOne
  stages: [Stage]
  state: StateType
  hectares: Float
  demo: Boolean
  yieldAdaptor: YieldAdaptor
  Layers: [Layer] @hasMany(indexName: "byProjectLayers", fields: ["id"])
  CalculatorScenarios: [CalculatorScenario] @hasMany(indexName: "byProjectCalculators", fields: ["id"])
  responses: [Response] @hasMany(indexName: "byProjectResponse", fields: ["id"])
}`

The full error I receive is:
`ERROR Error: Uncaught (in promise): Error: Namespace Resolver for 'Object' not found! This is probably a bug in '@amplify-js/datastore'.
Error: Namespace Resolver for 'Object' not found! This is probably a bug in '@amplify-js/datastore'.
    at IndexedDBAdapter.namespaceResolver (datastore.js:94:15)
    at IndexedDBAdapter.getIndexKeyValuesFromModel (IndexedDBAdapter.js:38:34)
    at IndexedDBAdapter.<anonymous> (IndexedDBAdapter.js:1031:42)
    at step (tslib.es6.js:102:1)
    at Object.next (tslib.es6.js:83:46)
    at fulfilled (tslib.es6.js:73:43)
    at _ZoneDelegate.invoke (zone.js:372:1)
    at Object.onInvoke (core.mjs:25608:1)
    at _ZoneDelegate.invoke (zone.js:371:1)
    at Zone.run (zone.js:134:1)
    at resolvePromise (zone.js:1211:1)
    at zone.js:1118:1
    at zone.js:1134:1
    at rejected (tslib.es6.js:74:89)
    at _ZoneDelegate.invoke (zone.js:372:1)
    at Object.onInvoke (core.mjs:25608:1)
    at _ZoneDelegate.invoke (zone.js:371:1)
    at Zone.run (zone.js:134:1)
    at zone.js:1275:1
    at _ZoneDelegate.invokeTask (zone.js:406:1)`

**Comment by madmaxaus** [comment]
> I have just started having this issue after updating @amplify-cli from v8 to v10.6. It is only happening on a specific model and when I try to delete. All other models are fine. I have changed the model schema and repushed using amplify-cli which did not fix the issue. The below is my model:
> 

I also am able to delete other models (i haven't tried them all). And I am also able to delete TestResults (offending model) via Amplify studio's content manager. (unfortunately, this is not a suitable workaround for my use case.)

**Comment by madmaxaus** [comment]
If i remove the 2 lines:

```
  UploadedFiles: [UploadedFiles] @hasMany(indexName: "byTestResults", fields: ["id"])
  TestResultData: [TestResultData] @hasMany(indexName: "byTestResults", fields: ["id"])
```

from my model deleting works.

For the record those models look like this:

```
type UploadedFiles @model @auth(rules: [{allow: public}, {allow: owner, operations: [create, read, update]}, {allow: groups, groups: ["admin"], operations: [read, create, update, delete]}]) {
  id: ID!
  CustomerID: ID! @index(name: "byCustomer")
  TestResultID: ID @index(name: "byTestResults")
  readUsers: [String]
  readGroups: [String]
  editUsers: [String]
  editGroups: [String]
  name: String
  s3: String
  identityId: String
  category: TestType
  date: AWSDateTime
  note: String
  status: FileTestStatus
}

type TestResultData @model @auth(rules: [{allow: public}, {allow: owner, operations: [create, read, update]}, {allow: groups, groups: ["admin"], operations: [read, create, update, delete]}]) {
  id: ID!
  name: String
  value: String
  unit: String
  testresultsID: ID! @index(name: "byTestResults")
  TestResultAnalyses: [TestResultAnalysis] @hasMany(indexName: "byTestResultData", fields: ["id"])
  testID: ID! @index(name: "byTest")
}

```

**Comment by chrisbonifacio** [comment]
Hi @madmaxaus @bobalucard 👋 thanks for raising this issue. Based on the description of the issue, we believe this may be related to #10864. We are already working on a fix for that issue and will ask you to verify whether it resolves this issue for you as well once released.

We appreciate your patience 🙏

**Comment by bobalucard** [comment]
@chrisbonifacio thank you for the update and quick response 😄

**Comment by iartemiev** [comment]
@madmaxaus, @bobalucard - we just released a fix for this in `aws-amplify@5.0.11`. Closing the issue as resolved. Please let us know if there's anything else we can help with.

**Comment by Bryson14** [comment]
I am getting this error with Amplify 11.0.4 still. 
This is my model below that causes me all the errors 
```
type Booking @model @auth(rules: [{allow: private}]) {
  id: ID!
  startDateTime: AWSDateTime!
  duration: Int!
  Tutor: Tutor! @hasOne
  bookingCode: String!
  amount: Float!
  discount: Float!
  clientNotes: String
  tutorNotes: String
  tutorPaid: Boolean!
  clientPaid: Boolean!
  resources: String
  Client: Client @hasOne
  studentName: String
  tutorRating: Int
  clientRating: Int
  bookingType: String
  tutorPayAmount: Float
  tip: Float
  transactionId: String
  tutorMileage: Float
  meetingUrl: AWSURL
  hwFiles: [String]
  resourceFiles: [String]
  meetingInstructions: String
  sessionTaught: Boolean
}
```
I will try to not use datastore and instead work with Appsync directly
